src/platforms/macos/daemon/macosdnsmanager.h

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef MACOSDNSMANAGER_H
+#define MACOSDNSMANAGER_H
+
+#include <systemconfiguration/scdynamicstore.h>
+
+#include <QMap>
+
+#include "command.h"
+
+// This class adds a command to the Mozilla VPN that overwrites the system
+// DNS configuration, and restores it upon program exit. This provides a
+// watchdog mechansim to ensure that no matter how the daemon termanates
```

Comment:
Some typos here. Also, could this say this it runs in a different process?

Reply:
The forking and separation of processes is actually handled by the `QProcess` class in `DnsUtilsMacos`, so I think it would be a bit misleading to describe it here.